exporters/jaeger/src/recordable.cc

```diff
@@ -193,7 +193,33 @@ void JaegerRecordable::SetInstrumentationLibrary(
 void JaegerRecordable::AddLink(const trace::SpanContext &span_context,
                                const common::KeyValueIterable &attributes) noexcept
 {
-  // TODO: convert link to SpanRefernece
+  // Note: "The Link’s attributes cannot be represented in Jaeger explicitly."
+  // -- https://opentelemetry.io/docs/reference/specification/trace/sdk_exporters/jaeger/#links
+  //
+  // This implementation does not (currently) implement the optional conversion to span logs.
+
+  thrift::SpanRef reference;
+
+  reference.__set_refType(thrift::SpanRefType::FOLLOWS_FROM);
+
+  // IDs should be converted to big endian before transmission.
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids
+#if JAEGER_IS_LITTLE_ENDIAN == 1
+  reference.__set_traceIdHigh(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()))));
+  reference.__set_traceIdLow(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1)));
+  reference.__set_spanId(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data()))));
+#else
+  reference.__set_traceIdLow(
+      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data())));
+  reference.__set_traceIdHigh(
+      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1));
+  reference.__set_spanId(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data())));
+#endif
```

Comment:
nit - can this code be shared between [JaegerRecordable::SetIdentity()](https://github.com/open-telemetry/opentelemetry-cpp/blob/c96a3e32f0b5bf17e58b1300383f9f205f33558f/exporters/jaeger/src/recordable.cc#L135) and here?

Reply:
I considered that, but there are two things that made me lean away from doing it:
1. The target objects are different (`thrift::Span` vs. `thrift::SpanRef`), so I think I'd need to do it with a macro instead of a function. And things are already fairly complex here (at least for my neophyte tastes), which made me avoid the macro. If there's a better way to do it, I'm happy to take pointers.
2. We need to do slightly different work in each case--`thrift::Span` needs the `parent_span_id` copied, but that doesn't exist on `thrift::SpanRef`.

Neither of them are showstoppers, so I'm happy to take a whack at merging it (except for the `parent_span_id`, of course).

I tried expanding the inline otel_bswap_64 function and/or adding a macro function to do a little bit more of the heavy lifting that happens on each line and ran into problems with both. In particular, the varying size of the data, how indexing changes, and where the high/low pieces go made it fairly complex.

The only thing I can think that would make sense would be moving the bits for trace ID and span ID into a macro that would be called like this.... and this works (at least the tests passed):
```
# define FOO(span_context, dst) \
  (dst)->__set_traceIdHigh( \
      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data())))); \
  (dst)->__set_traceIdLow( \
      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1))); \
  (dst)->__set_spanId( \
      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data()))));
#else
# define FOO(span_context, dst) \
  (dst)->__set_traceIdLow( \
      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()))); \
  (dst)->__set_traceIdHigh( \
      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1)); \
  (dst)->__set_spanId(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data())));
#endif
...
FOO(span_context, span_); // JaegerRecordable::SetIdentity
...
FOO(span_context, &reference); //  JaegerRecordable::AddLink
```

Reply:
Yes I can see the issue here, don't think this to be blocker. We can leave it for now.